src/main/java/ca/wolfram/beta/symath/MathUtils.java

```diff
@@ -19,4 +24,31 @@ public static void print(String s, Object... o) {
         System.out.println(String.format(Locale.CANADA, s, o));
     }
 
+    public static Set<MathNode> getVariableTerms(MathNode node) {
+        Set<MathNode> set;
+        switch (node.getType()) {
+        case CONSTANT:
+        case MATH_CONSTANT:
+            set = new HashSet<MathNode>();
+            break;
+        case VARIABLE:
+            set = new HashSet<MathNode>();
+            set.add(node);
+            break;
+        default:
+            set = node.getChildren().stream().filter(children -> !children.isConstant()).collect(Collectors.toSet());
```

Comment:
i don't want getVariableTerm to be recursive, because it no longer make sense after that. Say i have an addNode with multiply nodes with it. how does getting all the variable nodes help? plus if i do it like that, i can no longer use this function for the groupByLikeTerm function.  Another thing is, if the simplify is done correctly then all the like terms should have the same terms in the same format.